sonic_installer/main.py

```diff
@@ -389,8 +485,22 @@ def sonic_installer():
               help="Do not migrate current configuration to the newly installed image")
 @click.option('--skip-package-migration', is_flag=True,
               help="Do not migrate current packages to the newly installed image")
+@click.option('--setup-swap-if-necessary', is_flag=True,
+              help='Create temporary SWAP memory used for installation if necessary')
+@click.option('--swap-mem-size', default=1024, type=int, show_default='1024 MiB',
+              help='SWAP memory space size', callback=validate_positive_int,
+              cls=clicommon.SubOrdinateOption, depends_on=['setup_swap_if_necessary'])
+@click.option('--total-mem-threshold', default=2048, type=int, show_default='2048 MiB',
+              help='If system total memory is lower than threshold, setup SWAP memory',
+              cls=clicommon.SubOrdinateOption, depends_on=['setup_swap_if_necessary'],
+              callback=validate_positive_int)
+@click.option('--available-mem-threshold', default=1200, type=int, show_default='1200 MiB',
+              help='If system available memory is lower than threhold, setup SWAP memory',
+              cls=clicommon.SubOrdinateOption, depends_on=['setup_swap_if_necessary'],
+              callback=validate_positive_int)
 @click.argument('url')
-def install(url, force, skip_migration=False, skip_package_migration=False):
+def install(url, force, skip_migration=False, skip_package_migration=False,
+            setup_swap_if_necessary=False, swap_mem_size=None, total_mem_threshold=None, available_mem_threshold=None):
```

Comment:
Our default behavior should be trying to setup this swap mem when needed. You can add an option to disable the behavior.

Reply:
Sure, will change.